packages/flutter/lib/src/material/date_picker.dart

```diff
@@ -453,22 +463,28 @@ class _DatePickerDialogState extends State<DatePickerDialog> with RestorationMix
 
   @override
   Widget build(BuildContext context) {
-    final ThemeData theme = Theme.of(context);
-    final ColorScheme colorScheme = theme.colorScheme;
     final MaterialLocalizations localizations = MaterialLocalizations.of(context);
     final Orientation orientation = MediaQuery.orientationOf(context);
-    final TextTheme textTheme = theme.textTheme;
+    final DatePickerThemeData datePickerTheme = DatePickerTheme.of(context);
+    final DatePickerThemeData defaults = DatePickerTheme.defaults(context);
+    final bool useMaterial3 = Theme.of(context).useMaterial3;
```

Comment:
You could move the `Theme.of(context)` to a local final `theme` variable to avoid looking it up three times.

Reply:
Addressed in https://github.com/flutter/flutter/pull/119033/commits/70d860d720f506a57c730749eaaba252e4cd8a6b